src/main/java/org/broadleafcommerce/vendor/amazon/s3/S3FileServiceProvider.java

```diff
@@ -80,15 +90,42 @@ public File getResource(String name) {
 
     @Override
     public File getResource(String name, FileApplicationType fileApplicationType) {
-        File returnFile = blFileService.getLocalResource(buildResourceName(name));
+        File returnFile = blFileService.getLocalResource(getResourceName(name));
         OutputStream outputStream = null;
         InputStream inputStream = null;
 
         try {
             S3Configuration s3config = s3ConfigurationService.lookupS3Configuration();
             AmazonS3Client s3 = getAmazonS3Client(s3config);
-            S3Object object = s3.getObject(new GetObjectRequest(s3config.getDefaultBucketName(), buildResourceName(name)));
-            
+            String resourceName = getResourceName(name);
```

Comment:
I believe i see the shortcomings now. If i have the resource on the local filesystem, then there was no need for to look it up on the remote server.